### PR TITLE
attempted react-carousel-component

### PR DESCRIPTION
**Pull request by do-jonathan4**
F1

### CODE REVIEW

File: react-carousel-component/src/carousel.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    if (this.state.state.activeInterval) {
+      clearInterval(this.state.activeInterval)
+      this.setState({ activeInterval: null })
+    } else {
+      const newInterval = setInterval(this.changeIndex, 1000)
+      this.setState({ activeInterval: newInterval })
+    }
+  }
+  changeIndex(n) {
+    const slides = document.getElementsByClassName("slide");
```
Comment (scbowler) [inline react-carousel-component/src/carousel.jsx]: You don't want to query the DOM when using React. A better solution here would be to use `this.props.items`, you can check the length of that array to determine the next index.

File: react-carousel-component/src/carousel.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    this.setState({ imgIndex: n })
+
+    if (n > slides.length - 1) {
+      n = 0
+      this.setState({ imgIndex: n })
+    }
+    if (n < 0) {
+      n = slides.length - 1
+      this.setState({ imgIndex: n })
+    }
```
Comment (scbowler) [inline react-carousel-component/src/carousel.jsx]: This should be structured a little different by using else if statements so you don't have the extra setState call here on line 24.

```JSX
if (n >= slides.length) {
  ...
} else if (n < 0) {
  ...
} else {
  ...
}
```

**Review by scbowler — CHANGES_REQUESTED** [review]
Great job on this so far, a couple cleanup things needed. See below comments for details. Let me know if you have any questions.

**Comment by do-jonathan4** [comment]
F1

File: react-carousel-component/src/carousel.jsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        <img src={x.url}/>
+      </div>
+    )
+    const dot = this.props.items.map((x, i) =>
+      <div key={x.id}
+      className={`dot ${this.state.imgIndex === i ? 'active' : ''}`}
+        onClick={() => this.changeIndex(i)}
+      ></div>
+    )
+    window.onload = this.slideShow
```
Comment (scbowler) [inline react-carousel-component/src/carousel.jsx]: This is not the correct way to do this in React. Remember there are life cycle method in React components.

You want to call `this.slideShow()` from the `componentDidMount` method

**Review by scbowler — CHANGES_REQUESTED** [review]
Sooooo close, one more change. See below comment for details. Let me know if you have any questions.

**Review by scbowler — APPROVED** [review]
Great job! 👍

**Pull request merged**